src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -36,20 +36,16 @@ bool DoesNodeHaveGPUs(const NodeResources &resources) {
 }
 }  // namespace
 
-int64_t HybridPolicyWithFilter(const ResourceRequest &resource_request,
-                               const int64_t local_node_id,
-                               const absl::flat_hash_map<int64_t, Node> &nodes,
-                               float spread_threshold, bool force_spillback,
-                               bool require_available,
-                               std::function<bool(int64_t)> is_node_available,
-                               NodeFilter node_filter) {
+int64_t SchedulingPolicy::HybridPolicyWithFilter(
+    const ResourceRequest &resource_request, bool force_spillback, bool require_available,
+    std::function<bool(int64_t)> is_node_available, NodeFilter node_filter) {
   // Step 1: Generate the traversal order. We guarantee that the first node is local, to
   // encourage local scheduling. The rest of the traversal order should be globally
   // consistent, to encourage using "warm" workers.
   std::vector<int64_t> round;
-  round.reserve(nodes.size());
-  const auto local_it = nodes.find(local_node_id);
-  RAY_CHECK(local_it != nodes.end());
+  round.reserve(nodes_.size());
```

Comment:
Do you think it would be better to always iterate through `nodes_` instead of building this other copy? I find this structure a bit weird anyway, and it would make more sense now to try to keep the saved index consistent.

Reply:
Will do it in the separate round robin pr.